src/snowflake/snowpark/ml/transformer.py

```diff
@@ -501,3 +508,303 @@ def transform(self, df: DataFrame) -> DataFrame:
+    ) -> None:
+        super().__init__(input_cols, output_cols)
+        self.strategy = strategy
+        self._states_table_name = random_name_for_temp_object(TempObjectType.TABLE)
+        self._states_table_cols = (
+            [f"states_{input_col}" for input_col in self._input_cols]
+            if self._input_cols
+            else None
+        )
+        self.n_bins = n_bins
+        self._fitted = False
+
+    @Transformer.input_cols.setter
+    def input_cols(self, value: List[str]) -> None:
+        self._input_cols = value
+        self._states_table_cols = (
+            [f"states_{input_col}" for input_col in self._input_cols]
+            if self._input_cols
+            else None
+        )
+
+    def _uniform_fit(self, df: DataFrame) -> "KBinsDiscretizer":
+        df._describe(self.input_cols, stats=["max", "min"]).select(
+            [
+                object_agg("summary", input_col).as_(states_col)
+                for input_col, states_col in zip(
+                    self.input_cols, self._states_table_cols
+                )
+            ]
+        ).write.save_as_table(self._states_table_name, create_temp_table=True)
+        return self
+
+    def _quantile_fit(self, df: DataFrame) -> "KBinsDiscretizer":
+        self._states_table_name = {}
+        for input_col, output_col, states_col in zip(
+            self.input_cols, self.output_cols, self._states_table_cols
+        ):
+            df_quantile = None
+            self._states_table_name[input_col] = random_name_for_temp_object(
+                TempObjectType.TABLE
+            )
+            """
+            get the quantile partition of input columns, we will reduce 1 from the minimum of input column and add 1 to
+            the maximum of input column, then set it as lower bound and upper bound, so that every value in input column
+            falls into particular quantile.
+            df_quantile looks like this:
+            -------------------------------------------------
+            |"RES_A"  |"STATES_A_LOWER"  |"STATES_A_UPPER"  |
+            -------------------------------------------------
+            |1        |0.5               |1.5               |
+            |2        |1.5               |2.0               |
+            |3        |2.0               |3.0               |
+            |4        |3.0               |5.0               |
+            -------------------------------------------------
+            while input column is :
+            -------
+            |"A"  |
+            -------
+            |1.5  |
+            |2.0  |
+            |3.0  |
+            |4.0  |
+            -------
+            """
+            for i in range(self.n_bins):
+                df_temp = df.select(
+                    [
+                        lit(i + 1).as_(output_col),
+                        iff(
+                            lit(i) == lit(0),
+                            builtin("percentile_disc")(0).within_group(input_col) - 1,
```

Comment:
I suggest not using -1 or +1, but use `lit(-sys.maxsize)` and `lit(sys.maxsize)` directly